### PR TITLE
TINY-8277: Upgrade dev dependencies

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-8277

Description of Changes:

Upgraded the following dev dependencies:
- `typescript` 4.4.3 -> 4.5.2
- `grunt-contrib-concat` 1.0.1 -> 2.0.0
- `@tinymce/eslint-plugin` 1.9.2 -> 1.9.3
- `tsconfig-paths-webpack-plugin` 3.5.1 -> 3.5.2
-  `webpack-dev-server` 3.11.2 -> 3.11.3

and the following oxide specific dev dependencies:
- `grunt-contrib-less` 2.0.0 -> 3.0.0
- `gulp-less` 4.0.1 -> 5.0.0
- `gulp-sourcemaps` 2.6.4 -> 3.0.0
- `stylelint-order` 4.0.0 -> 5.0.0

Note: Upgrading the less deps also upgraded  less 3.x -> less 4.x. I have compared the skin and content css output and it's the same.

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts [hunk 1]
```diff
@@ -121,8 +121,8 @@ const htmlDiff: (v1: string, v2: string) => string = (() => {
       }
     } else {
       if (out.n[0].text == null) {
-        for (n = 0; n < out.o.length && out.o[n].text == null; n++) {
-          str += '<del>' + out.o[n] + oSpace[n] + '</del>';
+        for (let j = 0; j < out.o.length && out.o[j].text == null; j++) {
```
Comment (lnewson) [inline modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:124]: TypeScript 4.5 complained about this as it was able to infer the original type for `n` was a string apparently :shrug: 

```
modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:124:14 - error TS2322: Type 'number' is not assignable to type 'string'.
124         for (n = 0; n < out.o.length && out.o[n].text == null; n++) {
                 ~

modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:124:64 - error TS2356: An arithmetic operand must be of type 'any', 'number', 'bigint' or an enum type.
124         for (n = 0; n < out.o.length && out.o[n].text == null; n++) {
                                                                   ~

modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:136:81 - error TS2356: An arithmetic operand must be of type 'any', 'number', 'bigint' or an enum type.
136           for (n = out.n[i].row + 1; n < out.o.length && out.o[n].text == null; n++) {
                                                                                    ~
```

Comment (TheSpyder) [reply modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:124]: Or maybe it's just more strict about not allowing changes to the type of a `let` variable

**Review by TheSpyder — APPROVED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -5624,10 +5595,10 @@ glob-watcher@^5.0.3:
     just-debounce "^1.0.0"
     object.defaults "^1.1.0"
 
-glob@^7.0.3, glob@^7.0.5, glob@^7.1.1, glob@^7.1.2, glob@^7.1.3, glob@^7.1.4, glob@^7.1.6, glob@~7.1.1, glob@~7.1.6:
-  version "7.1.7"
-  resolved "https://registry.yarnpkg.com/glob/-/glob-7.1.7.tgz#3b193e9233f01d42d0b3f78294bbeeb418f94a90"
-  integrity sha512-OvD9ENzPLbegENnYP5UUfJIirTg4+XwMWGaQfQTY0JenxNvvIKP3U3/tAQSPIu/lHxXYSZmpXlUHeqAIdKzBLQ==
+glob@^7.0.3, glob@^7.1.1, glob@^7.1.2, glob@^7.1.3, glob@^7.1.4, glob@^7.1.6:
+  version "7.2.0"
```
Comment (ashfordneil) [inline yarn.lock:5599]: Should this be deduplicated, and brought back to all depending on 7.1.7?

**Review by ashfordneil — APPROVED** [review]

File: yarn.lock (hunk 2 above)
Comment (lnewson) [reply yarn.lock:5599]: I've run `yarn-deduplicate` so this is fine... it's primarily only packages used in the dist that we cannot have duplicates for.

**Pull request merged**